src/cplscheme/BaseCouplingScheme.hpp

```diff
@@ -545,6 +463,99 @@ class BaseCouplingScheme : public CouplingScheme {
 
   /// Writes out coupling convergence within all time windows.
   std::shared_ptr<io::TXTTableWriter> _convergenceWriter;
+
+  /// First participant name.
+  std::string _firstParticipant = "unknown";
+
+  /// Second participant name.
+  std::string _secondParticipant = "unknown";
+
+  /// Local participant name.
+  std::string _localParticipant = "unknown";
+
+  /// Smallest number, taking validDigits into account: eps = std::pow(10.0, -1 * validDigits)
+  const double _eps;
+
+  /**
+   * @brief All convergence measures of coupling iterations.
+   *
+   * Before initialization, only dataID and measure variables are filled. Then,
+   * the data is fetched from send and receive data assigned to the cpl scheme.
+   */
+  std::vector<ConvergenceMeasure> _convergenceMeasures;
+
+  /// Sets whether the solver evaluates the fine or the coarse model representation
+  bool _isCoarseModelOptimizationActive = false;
+
+  /// Functions needed for initialize()
+  void setupConvergenceMeasures();
+
+  /// TODO
+  virtual void mergeData() = 0;
+
+  /// implements functionality for initialize in base class.
+  virtual void initializeImplementation() = 0;
+
+  /// Functions needed for initializeData()
+
+  /// implements functionality for initializeData in base class.
+  virtual void exchangeInitialData() = 0;
+
+  /// Functions needed for advance()
+
+  /// implements functionality for advance in base class.
+  virtual std::pair<bool, bool> exchangeDataAndAccelerate() = 0;
+
+  virtual DataMap &getAcceleratedData() = 0;
```

Comment:
Wouldn't "getAccelerationData" be a better name? The data is in general not accelerated yet.

Reply:
I agree. Fixed in c815df7b6c396b9b1ab6b6cdee30f59ee9902b7f.